settings.js

```diff
@@ -203,6 +203,11 @@ function generate(){
         // more generated below
       },
       "char_filter": {
+        "ampersand_splitter": {
+          "type": "pattern_replace",
+          "pattern": "&",
+          "replacement": " and "
```

Comment:
The order these are executed is:
1. "char_filter" rules are run on the strings before they are split into words
2. "tokenizer" runs next splitting on delimiter pattern (whitespace etc)
3. "filter" rules are then run on individual tokens

so that means you can get away with simply replacing  '&' with ' & '.

you shouldn't need to use the English form "and" here, the benefit of this is you don't need to make any changes to 'synonyms/punctuation/ampersand.txt' because it'll 'just work' 🧙 

A pattern of "&" will expand something like "Johnson & Johnson" to "Johnson  &  Johnson" (double spaces), this isn't probably a big deal but I think you can probably tighten up the pattern a bit by only matching an ampersand sandwiched between two characters, such as:

```
[a-zA-Z]&[a-zA-Z]

[^\s]&[^\s]
```

something like that

Reply:
You can probably prototype a fix for the "Mc Williams" issue using a similar technique

```
"pattern": "(mc)\s+([^\s])"
"replace": "$1$2"
```

one thing to be aware of with this is that it's running *before* lowercase, so it's case-sensitive whereas the token filters which later are case-insensitive.

this would have the effect of removing any spaces immediately following the characters "mc".